File: Fortran/OpenMP_Fortran_BASIC/example_3_OpenMP_Hello_World_Subroutine.f90
! Compile
! $ gfortran -fopenmp -o out example_2_OpenMP_Hello_World_Subroutine.f90 
! Set environment variable 
! $ export OMP_NUM_THREADS=16
! Execute 
! $ ./out 
!
! The !$OMP PARALLEL and !$OMP END PARALLEL directives creates a section of code that is run from all available threads.
 program hello_world
   use omp_lib
   implicit none
   integer threads_id 
!$omp parallel 
   threads_id = omp_get_thread_num() ! obtain thread number
   call hello(threads_id) 
!$omp end parallel

end program

subroutine hello(tid)
   integer tid
   write(*,*) 'hello world !!!, from thread = ', tid
end subroutine 

File: C/OpenMP_C_BASIC/example_1_Hello_World/example_3_OpenMP_Hello_World_Subroutine.c
/******************************************************************************
The "pragma omp parallel" directives creates a section of code that is run from all available threads.
Compile

 $ gcc -fopenmp -o out example_2_OpenMP_Hello_World_Subroutine.c
Set environment variable 
 $ export OMP_NUM_THREADS=16
Execute
 $ ./out
---------------------------------------------------------------------*/
#include <stdio.h>
#include <omp.h>  

int hello(int );

int main() {
   int threads_id;
  #pragma omp parallel 
  {
      threads_id = omp_get_thread_num(); // obtain thread number
      hello( threads_id );
   }
   return 0;
/*-------------------------------------------------------------------*/
int hello(int tid){
    printf( "Hello, World !!!, From thread %d!\n", tid);
}
